internal/schemautil/custom_diff.go

```diff
@@ -173,3 +175,70 @@ func CustomizeDiffCheckStaticIpDisassociation(_ context.Context, d *schema.Resou
 	// TODO: Check that we block deletions that will result in too few static ips for the plan
 	return nil
 }
+
+// CustomizeDiffDisallowMultipleManyToOneKeys checks that we don't have multiple keys that are going to be converted to
+// a single key in the API request, e.g. 'ip_filter' and 'ip_filter_object' in the same diff.
+func CustomizeDiffDisallowMultipleManyToOneKeys(_ context.Context, d *schema.ResourceDiff, _ interface{}) error {
+	ks := d.GetRawConfig().AsValueMap()
+
+	tks := map[string]struct{}{}
+
+	for k, v := range ks {
+		if !strings.Contains(k, "_user_config") {
+			continue
+		}
+
+		nks := v.AsValueSlice()[0].AsValueMap()
+
+		for k, v := range nks {
+			if v.IsNull() {
+				continue
+			}
+
+			if v.CanIterateElements() && len(v.AsValueSlice()) == 0 {
+				continue
+			}
+
+			ak := k
+
+			// TODO: Remove this in the next major version.
+			if k == "ip_filter" || k == "namespaces" {
+				ak = fmt.Sprintf("%s_string", k)
+			}
+
+			if userconfig.IsKeyTyped(ak) {
```

Comment:
We only explore the first level of user config, but we may have fields that we are looking for deeper, like `namespaces`

Reply:
fixed